streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -252,5 +573,72 @@ private void waitUntilAtLeastOneRecordProcessed() throws InterruptedException {
         return Collections.emptySet();
     }
 
+    private Map<String, Long> fetchMap(final ReadOnlyWindowStore<String, Long> store,
+                                       final String key) {
+
+        final WindowStoreIterator<Long> fetch = store.fetch(key, 0, System.currentTimeMillis());
+        if (fetch.hasNext()) {
+            KeyValue<Long, Long> next = fetch.next();
+            return Collections.singletonMap(key, next.value);
+        }
+        return Collections.emptyMap();
+    }
+
+
+    /**
+     * A class that periodically produces records in a separate thread
+     */
+    private class ProducerRunnable implements Runnable {
+        private String topic;
+        private final List<String> inputValues;
+        private final int numIterations;
+        private int currIteration = 0;
+        boolean shutdown = false;
+        private final Random random = new Random();
+
+        ProducerRunnable(String topic, List<String> inputValues, int numIterations) {
+            this.topic = topic;
+            this.inputValues = inputValues;
+            this.numIterations = numIterations;
+        }
+
+        public synchronized void incrementInteration() {
+            currIteration++;
+        }
+        public synchronized int getCurrIteration() {
+            return currIteration;
+        }
+        public synchronized void shutdown() {
+            shutdown = true;
+        }
+
+        @Override
+        public void run() {
+            Properties producerConfig = new Properties();
+            producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+            producerConfig.put(ProducerConfig.ACKS_CONFIG, "all");
+            producerConfig.put(ProducerConfig.RETRIES_CONFIG, 0);
+            producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+            producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+            final KafkaProducer<String, String>
+                producer =
+                new KafkaProducer<>(producerConfig, new StringSerializer(), new StringSerializer());
+
+            try {
+                while (getCurrIteration() < numIterations && !shutdown) {
+                    for (int i = 0; i < inputValues.size(); i++) {
+                        producer.send(new ProducerRecord<>(topic,
```

Comment:
We do not need to set the key since it can be null: 

```
    public ProducerRecord(String topic, V value) {
        this(topic, null, null, null, value);
    }
```